File: backend/src/main/java/com/softdight/instantorder/backend/model/SubMenu.java
package com.softdight.instantorder.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "SUBMENU", catalog = "MENU_SCHEMA")
public class SubMenu extends RestaurantDependent{

    @Column(name = "PATH_TO_PHOTO")
    private String pathToPhoto;

    /*
    * Each submenu have one or more categories
    * */
    @OneToMany(mappedBy = "subMenu", cascade = CascadeType.ALL)
    private Set<SubCategorySubMenu> categories;

    @OneToMany(mappedBy = "subMenu", cascade = CascadeType.ALL)
    private Set<SubMenuRecipe> recipes;

    public Recipe getRecipeById(String recipeId){
        for(SubMenuRecipe subMenuRecipe : getRecipes()){
            if(subMenuRecipe.getRecipe().getId().equals(recipeId)){
                return subMenuRecipe.getRecipe();
            }
        }
        return null;
    }

}

File: Frontend/App/VirtualWaiter/Assets/Scripts/DataModel/SubMenu.cs
using System.Collections.Generic;

public class SubMenu : EntityBaseModel {

    public string PathToPhoto;

    public List<Recipe> Recipes;

    public List<OptionalRecipe> OptionalRecipes;

    public SubMenu() : base()
    {
        Recipes = new List<Recipe>();
        OptionalRecipes = new List<OptionalRecipe>();
    }

    public SubMenu(string id, string pathToPhoto, List<Recipe> recipes) : this()
    {
        base.id = id;
        Recipes = new List<Recipe>();
        PathToPhoto = pathToPhoto;
        OptionalRecipes = new List<OptionalRecipe>();
    }

    public SubMenu(string id, string pathToPhoto, List<Recipe> recipes, List<OptionalRecipe> optionalRecipes)
    {
        base.id = id;
        Recipes = recipes;
        PathToPhoto = pathToPhoto;
        OptionalRecipes = optionalRecipes;
    }
}
